packages/devtools-source-map/src/source-map.js

```diff
@@ -182,6 +189,140 @@ async function getOriginalLocation(location: Location) : Promise<Location> {
   };
 }
 
+function _comparePositions(a: SourceMapPosition, b: SourceMapPosition): number {
+  if (a.line === b.line) {
+    return a.column - b.column;
+  } else {
+    return a.line - b.line;
+  }
+}
+
+function _getOriginalPositionName(
+  map: SourceMapConsumer,
+  generatedPosition: SourceMapPosition,
+): (string|null) {
+  const originalPosition = map.originalPositionFor(generatedPosition);
+  if (!originalPosition || !originalPosition.name) {
+    return null;
+  }
+  // Verify if it's the exact position.
+  const originalPositionTest = map.originalPositionFor({
+    line: generatedPosition.line,
+    column: generatedPosition.column,
+    bias: SourceMapConsumer.LEAST_UPPER_BOUND
+  });
+  if (
+    !originalPositionTest ||
+    originalPositionTest.column !== originalPosition.column ||
+    originalPositionTest.line !== originalPosition.line
+  ) {
+    return null;
+  }
+  return originalPosition.name;
+}
+
+async function getLocationScopes(
```

Comment:
I think `getLocationScope` is sufficiently complex to justify its own module with some helper methods.

```js

function appendTrackedBinding() {}

function appendBinding() {}

function getScopeBindings() {}

function getLocationScope() {

  generatedSourceScopes.map(scope => {
    // ...
    const trackedBindings = bindings.reduce(appendTrackedBinding, {})
    const bindings = trackedBindings.reduce(appendBinding, {});
    return {
      type: scope.type,
      bindings,
    };
  })
}

module.exports = { getLocationScope}
```

Reply:
appendTrackedBinding and appendBinding extraction is not possible due to dependency on location/currentPosition.

Reply:
getLocationScope also depends on _getSourceMap

Reply:
we might need to do a bit more splitting to share `_getSourceMap` between modules.

we also might need to pass in the closure state:

```js
bindings.reduce(binding => appendTrackedBinding(binding, location, ...), {})
```

Reply:
Did not extracted _getSourceMap -- just exposed it from ./source-map

Moved scopes into separate file and refactored to have appendTrackedBinding.